### PR TITLE
Rename Analyze by Explain

### LINKED ISSUE #1586: Rename Analyze in package sql/ir into Explain
https://godoc.org/github.com/sql-machine-learning/sqlflow/pkg/sql/ir

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1586

### CODE REVIEW

File: pkg/sql/codegen/xgboost/template_analyze.go [hunk 1]
```diff
@@ -17,15 +17,15 @@ import (
 	"text/template"
```
Comment (shendiaomo) [inline pkg/sql/codegen/xgboost/template_analyze.go:17]: Rename this file to `template_explain.go`?

File: pkg/sql/codegen/xgboost/template_analyze.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ label_name = "{{.Label}}"
 
 summaryAttrs = json.loads('''{{.ShapSummaryParams}}''')
 
-def analyzer_dataset():
+def explainr_dataset():
```
Comment (shendiaomo) [inline pkg/sql/codegen/xgboost/template_analyze.go]: explainr -> explainer?

**Review by shendiaomo — COMMENTED** [review]
There is still `Analyze` in `main_test.go`

**Comment by wangkuiyi** [comment]
Thanks to @shendiaomo for fixing the issue in https://github.com/sql-machine-learning/sqlflow/pull/1590. Close this one.

**Pull request closed without merging**